test/refract-test.coffee

```diff
@@ -331,5 +331,28 @@ describe('Transformations • Refract', ->
         assert.equal(ast.sections[1].resources.length, 2)
       )
     )
+
+    describe('Mixed Resources and Resource Groups', ->
+      applicationAst = null
+
+      before(->
+        applicationAst = convertToApplicationAst(
+          require('./fixtures/refract-parse-result-tags.json')
+        )
+      )
+
+      it('Has two sections', ->
+        assert.strictEqual(applicationAst.sections.length, 2)
+      )
+
+      it('First section contains two resources', ->
+        assert.strictEqual(applicationAst.sections[0].resources.length, 6)
+      )
+
+      it('Second section contains two resources', ->
```

Comment:
@smizell 

Good point, fixed in https://github.com/apiaryio/metamorphoses/commit/2e1b36a3fccd5337278ce4ed57d089679eb85b16.